### PR TITLE
feat: Kg/explore expanding search bar

### PR DESCRIPTION
**Pull request by kayleegeorge**
expanding search bar with X functionality

https://user-images.githubusercontent.com/62825936/179059937-aad26d89-7d7a-4a19-ada4-d15e2b8c3190.mov

### CODE REVIEW

File: src/components/Explore/SearchBar.tsx [hunk 1]
```diff
@@ -38,16 +50,42 @@ const SearchInput = styled.input`
...
   const [isFocused, setFocused] = useState(false)
+  const [isExpanded, setExpanded] = useState(false)
+  const [searchContent, setSearchContent] = useState('')
+  const searchRef = useRef<HTMLInputElement>(null)
+
+  const clickSearchIcon = () => {
+    setExpanded(true)
+    setFocused(true)
+
+    if (searchRef.current) {
```
Comment (kayleegeorge) [inline src/components/Explore/SearchBar.tsx:61]: @vm this runs

File: src/components/Explore/SearchBar.tsx [hunk 2]
```diff
@@ -1,54 +1,85 @@
...
+
+const SearchInput = styled.input<{ expanded: boolean }>`
+  background: no-repeat scroll 7px 7px;
+  background-image: url(${searchIcon});
+  background-size: 20px 20px;
+  background-position: 12px center;
+  background-color: ${({ theme }) => theme.bg0};
+  left: 10px;
+  border-radius: 12px;
+  align-items: center;
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:23]: is this needed?

File: src/components/Explore/SearchBar.tsx [hunk 3]
```diff
@@ -1,54 +1,85 @@
...
   border: none;
-  width: 90%;
+  height: 100%;
+  width: ${({ expanded }) => (expanded ? '100%' : '44px')};
   font-size: 16px;
+  padding-left: 40px;
   color: ${({ theme }) => theme.text2};
+  animation: ${({ expanded }) => expanded && 'expand 0.8s'};
+  @keyframes expand {
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:31]: i would add a transition for the bg color

File: src/components/Explore/SearchBar.tsx [hunk 4]
```diff
@@ -1,54 +1,85 @@
...
   }
   ::placeholder {
-    color: ${({ theme }) => theme.text3};
+    color: ${({ theme, expanded }) => (expanded ? theme.text3 : 'transparent')};
     @media only screen and (max-width: ${SMALL_MEDIA_BREAKPOINT}) {
       color: transparent;
     }
   }
+  ::-webkit-search-cancel-button {
+    -webkit-appearance: none;
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:51]: dont need this

File: src/components/Explore/SearchBar.tsx [hunk 5]
```diff
@@ -1,54 +1,85 @@
...
+    appearance: none;
+    height: 24px;
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:53]: you need the height, width and background size? i think make a var for the 24 tho

File: src/components/Explore/SearchBar.tsx [hunk 6]
```diff
@@ -1,54 +1,85 @@
...
 `
 
 export default function SearchBar() {
-  const theme = useTheme()
-  const [isFocused, setFocused] = useState(false)
+  const [isExpanded, setExpanded] = useState(false)
+  const [searchContent, setSearchContent] = useState('')
+  const searchRef = useRef<HTMLInputElement>(null)
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:66]: usecallback

Comment (vm) [reply src/components/Explore/SearchBar.tsx:66]: also `onSearchChange`

File: src/components/Explore/SearchBar.tsx (hunk 5 above)
Comment (kayleegeorge) [reply src/components/Explore/SearchBar.tsx:53]: for some reason yes, when i remove either/or, the framing gets weird haha

File: src/components/Explore/SearchBar.tsx [hunk 7]
```diff
@@ -1,54 +1,85 @@
...
+    setSearchContent(event.target.value)
+  }
 
   return (
-    <StyledSearchBar focused={isFocused}>
-      <Search size={20} color={theme.text3} />
+    <SearchBarWrapper>
```
[the remark this answers is not in the document]
Comment (kayleegeorge) [reply src/components/Explore/SearchBar.tsx:71]: oh the relative positioning was for when i twas trying to make the search icon absolute -- removed now

**Comment by JFrankfurt** [comment]
The searchbar is currently rendering in at full size, then animating to the collapsed state immediately. Can you set it to be in whichever state it's supposed to start with so it's not instantly animating on page load?

**Comment by kayleegeorge** [comment]
> looks like you've got some browser-compatibility issues:
> 
> left is firefox, right is safari image

Safari seems to work for me now -- firefox doesn't support webkit cancel button on the input of type search sadly

**Comment by vm** [comment]
X Icon isn't centered in safari. No icon in firefox I think is okay.

File: src/components/Explore/SearchBar.tsx [hunk 8]
```diff
@@ -1,60 +1,74 @@
...
+  background-image: url(${searchIcon});
+  background-size: 20px 20px;
+  background-position: 12px center;
+  background-color: ${({ theme }) => theme.bg0};
+  left: 10px;
+  border-radius: 12px;
   border: none;
-  width: 90%;
+  height: 100%;
+  width: ${({ expanded }) => (expanded ? '100%' : '44px')};
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:27]: shouldn't need to do this, just have it be width: 100 on focused state

File: src/components/Explore/SearchBar.tsx [hunk 9]
```diff
@@ -1,60 +1,74 @@
...
-        type="text"
-        value={filterString}
-        onChange={({ target: { value } }) => setFilterString(value)}
+        expanded={isExpanded}
+        type="search"
         placeholder="Search token or paste address"
         id="searchBar"
-        onFocus={() => setFocused(true)}
-        onBlur={() => setFocused(false)}
+        onBlur={() => isExpanded && filterString.length === 0 && setExpanded(false)}
```
Comment (vm) [inline src/components/Explore/SearchBar.tsx:66]: shouldn't need to manually setExpanded

**Review by vm — CHANGES_REQUESTED** [review]
comments above, otherwise good

File: src/components/Explore/SearchBar.tsx (hunk 9 above)
Comment (kayleegeorge) [reply src/components/Explore/SearchBar.tsx]: I think I need isExpanded for the check on whether filterstring is size 0 (if filterstring isn't of length 0 then it shouldn't collapse)

**Review by vm — APPROVED** [review]
fiiiiiire

**Pull request merged**